src/main/java/com/redhat/exhort/utils/Environment.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright © 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.exhort.utils;
+
+import java.util.Optional;
+
+public final class Environment {
+
+  private Environment() {}
+
+  public static final String get(String name, String defaultValue) {
+    return Optional.ofNullable(System.getProperty(name))
+        .or(() -> Optional.ofNullable(System.getenv(name)))
+        .orElse(defaultValue);
+  }
+
+  public static final String get(String name) {
+    return get(name, null);
+  }
+
+  public static final boolean getBoolean(String key, Boolean defaultValue) {
```

Comment:
non-blocking nit: should `defaultValue` be `Boolean` vs primitive type? If someones going to pass `null` Id imagine they might as well pass `false` so things are clearer at call-site

Reply:
You're right. That's a refactoring leftover. Thanks